### DISCUSSION TITLE
Support for Hybrid Flow

### DISCUSSION THREAD

**Discussion by ilyashtrikul** [zitadel/zitadel#5066] [General]
Do you have plans to support Hybrid Flow in the Zitadel? Zitadel is all the best from Auth0 and Keycloak - but this functionality is missing. For now I really need `code id_token` response type and I was confused when I didn't find it in Zitadel.

**Comment by fforootd** [comment]
Hm, in what case/language/framework are you using hybrid at the moment?

**Reply by ilyashtrikul** [reply]
This flow is used by a third party company that provides services to us. They are a client for our oAuth2 server and can only work on this mechanism. They use the same flow for both the web and the mobile application and do not want to modify the client, they explain that the Hybrid Flow is safer. I don't know exactly what stack they use, most likely it's PHP and JavaScript.

**Reply by fforootd** [reply]
Ok thank you for the extra insights.

>  they explain that the Hybrid Flow is safer

Hm I would not agree on this end. Currently there is a agreement that "only" the authorization code flow (incl. PKCE) is "secure".
That is the reason why we currently do not support these flows.

We would certainly be open for contributions on this end 😬

**Reply by ilyashtrikul** [reply]
Thank you for your answer. I have read the Hybrid Flow [documentation](https://auth0.com/docs/get-started/authentication-and-authorization-flow/hybrid-flow). It says that in this case the Authentication Code Flow is used and the Implicit Flow is additionally used. Can we conclude from this that there is a higher level of security?

**Comment by ilyashtrikul** [comment]
And as I understand

> Because regular web apps are server-side apps where the source code is not publicly exposed, they can use the Authorization Code Flow (defined in [OAuth 2.0 RFC 6749, section 4.1](https://tools.ietf.org/html/rfc6749#section-4.1)), which exchanges an Authorization Code for a token

Authorization Code Flow is more suitable for server-side applications, when Hybrid Flow is suitable for native applications and server-side rendered clients, such as Web APIs and native mobile applications.

**Reply by fforootd** [reply]
Yes that was true in the past, nowadays even single page apps and public clients in general rely on the authorization code flow with pkce. 

One of reasons is to better protect clients against mitm attacks and callback compromising.
